hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -228,6 +228,26 @@ public Builder append(String key, String value) {
       return this;
     }
 
+    /**
+     * Append new field which contains key and value to the context
+     * if the key("key:") is absent.
+     * @param key the key of field.
+     * @param value the value of field.
+     * @return the builder.
+     */
+    public Builder appendIfAbsent(String key, String value) {
```

Comment:
As this is pretty core; I would prefer to do the changes to commons in a HADOOP JIRA.

Reply:
> As this is pretty core; I would prefer to do the changes to commons in a HADOOP JIRA.

Thanks @goiri for your review and comments. Do I need to close the issue and file a Hadoop Jira? Looking forward to your reply. Thanks a lot.

Reply:
I would create a new one just for the commons side and once that's done, rebase this one to do just the RBF part.

Reply:
Thank you very much for your advice and I will create a new one.

Reply:
> As this is pretty core; I would prefer to do the changes to commons in a HADOOP JIRA.

Hi @goiri , I filed a HADOOP JIRA [HADOOP-18003](https://issues.apache.org/jira/browse/HADOOP-18003) and the relevant PR has been merged. I rebased this PR. Could you please help review the code. Thank you very much.